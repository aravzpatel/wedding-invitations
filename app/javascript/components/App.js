import React from "react"
import Welcome from "./Welcome"
import PropTypes from "prop-types"
import Artwork from "./Artwork"
import Button from "./Button"
class App extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      displayWelcome: true
    }
  }
  render () {
    let artwork = (
      <div>
        <Artwork />
      </div>
    )
    let welcome = (
      <div>
        <Welcome guests={this.props.guests}/>
      </div>
    )
    
    let button = (
      <div>
        <Button />
      </div>
    )
    
    if(this.state.displayWelcome === false){
      artwork = null
    }

    return (
      <div>
        {welcome}
        {artwork}
        {button}
      </div>
    );
  }
}

App.propTypes = {
  guests: PropTypes.array
};

export default App
