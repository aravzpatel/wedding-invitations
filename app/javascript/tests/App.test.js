import React from 'react';
import App from '../components/App';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';


it('App renders with welcome message', () => {
  const guests = [{id: 1, first_name: 'Jack', last_name: 'Harper'}, {id: 2, first_name: 'Jill', last_name: 'Harper'}]
  const component = renderer.create(
    <App guests = {guests}/>
  )

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})

it('App re-renders with correct forms', () => {
  const guests = [{id: 1, first_name: 'Jack', last_name: 'Harper'}, {id: 2, first_name: 'Jill', last_name: 'Harper'}]
  const component = shallow(
    <App guests = {guests}/>
  )

  component.instance().changeView();

  let tree = toJson(component);
  expect(tree).toMatchSnapshot();

  component.instance().onRSVPSubmit()
  component.instance().onRSVPSubmit()

  expect(toJson(component)).toMatchSnapshot();

  component.instance().onAddressSubmit()
  expect(toJson(component)).toMatchSnapshot();

})

