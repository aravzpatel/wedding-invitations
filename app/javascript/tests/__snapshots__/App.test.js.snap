// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App re-renders with correct forms 1`] = `
<Container
  fluid={false}
>
  <div
    className="form-container"
  >
    <div>
      <RsvPform
        guests={
          Array [
            Object {
              "first_name": "Jack",
              "id": 1,
              "last_name": "Harper",
            },
            Object {
              "first_name": "Jill",
              "id": 2,
              "last_name": "Harper",
            },
          ]
        }
        onSubmit={[Function]}
      />
    </div>
  </div>
</Container>
`;

exports[`App re-renders with correct forms 2`] = `
<Container
  fluid={false}
>
  <div
    className="form-container"
  >
    <div>
      <AddressForm
        onSubmit={[Function]}
      />
    </div>
  </div>
</Container>
`;

exports[`App re-renders with correct forms 3`] = `
<Container
  fluid={false}
>
  <div
    className="form-container"
  >
    <div>
      <Thankyou />
    </div>
  </div>
</Container>
`;

exports[`App renders with welcome message 1`] = `
<div
  className="container"
>
  <div
    className="form-container"
  >
    <div>
      <div>
        <div
          className="artwork-container"
        >
          <img
            alt="Image"
            className="img-fluid"
            src="ArtworkMock"
          />
          <div
            className="welcome-greeting"
          >
            <div>
              Jack and Jill
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
